Switch CI to GH Actions, get build green again

palango opened the pull request:
### What this PR does

- Switch CI to GH Actions, as Travis was really slow with builds starting only after several minutes wait.
- Get the build green again, disable the new `pip` resolver
- Update `black`

#### Missing
- The release pipeline is not automated yet. I can add this, but thought this wouldn't be important in a first step.
- Travis checks should be removed, but I have no admin rights

### Why I'm making this PR

### What's tricky about this PR (if any)

----

Any reviewer can check these:

* [ ] If the PR is fixing a bug or adding a feature, add an entry to CHANGELOG.md.
* [ ] If the PR changed a Solidity source, run `make compile_contracts` and add the resulting `raiden_contracts/data/contracts.json` in the PR.
* [ ] If the PR is changing documentation only, add `[skip ci]` in the commit message so Travis does not waste time.
    * [ ] But, if the PR changes comments in a Solidity source, do not add `[skip ci]` and let Travis check the hash of the source.
* [ ] In Python, use keyword arguments
* [ ] Squash unnecessary commits
* [ ] Comment commits
* [ ] Follow naming conventions
    * `solidityFunction`
    * `_solidity_argument`
    * `solidity_variable`
    * `python_variable`
    * `PYTHON_CONSTANT`
* [ ] Follow the Signature Convention in [CONTRIBUTING.md](./CONTRIBUTING.md)
* For each new contract
    * [ ] The deployment script deploys the new contract.
    * [ ] `etherscan_verify.py` runs on the new contract.
* Bookkeep
    * [ ] The gas cost of new functions are stored in `gas.json`.
* Solidity specific conventions
    * [ ] Document arguments of functions in natspec
    * [ ] Care reentrancy problems
    * if the PR adds or removes `require()` or `assert()`
        * [ ] add an entry in Changelog
        * [ ] open an issue in the client, light client, service repos so the change is reflected there
        * Just adding a message in `require()` doesn't require these steps.
* [ ] When you catch a require() failure in Solidity, look for a specific error message like `pytest.raises(TransactionFailed, match="error message"):`

And before "merge" all checkboxes have to be checked.  If you find redundant points, remove them.

karlb reviewed (approved):
Finally no repo with Travis left. Thanks!

I removed travis from the branch protection rule and added you as an admin. I was not sure which status checks should be enforced now. Is it ""build (3.7)" and " "build (3.8)"?

palango commented:
> I removed travis from the branch protection rule and added you as an admin. I was not sure which status checks should be enforced now. Is it ""build (3.7)" and " "build (3.8)"?

I think it should be the `lint-and-test (3.X)` ones, but I have no clue what the build 3.X`ones are.
